mm0-rs/src/elab/lisp/debug.rs

```diff
@@ -0,0 +1,1465 @@
+//! Debug formatting for mm0-rs items. Follows the API set out by the `print` and `pretty`
+//! modules. 
+use std::collections::HashMap;
+use std::cell::RefCell;
+use std::{fmt::{self}};
+use shoebill::{ Printer, Doclike, DocPtr };
+use shoebill::ron::{ RonStruct, RonSequence, RonTuple, RonOption, RonResult };
+use super::{LispVal, LispKind, print::FormatEnv,
+  super::{
+    environment::{DeclKey, NotaInfo, AtomData, AtomID, SortID, TermID, ThmID, Term, Thm, Type, Sort, Expr }, }};
+use crate::util::{ FileSpan, FileRef, Span };
+
+/// Debug counterpart to the `Print` struct`.
+pub struct DebugPrint<'a, D : ?Sized> {
```

Comment:
This struct is the same as `Print`. How about implementing `Debug` for `Print` instead of this?

(PS: No space before colon.)